api/v2/Routes/isadmin/index.js

```diff
@@ -6,13 +6,24 @@ const router = Router({ mergeParams: true });
 
 // Responds with whether or not the current user is an admin
 router.get('/', async (req, res) => {
-    const authHeader = req.headers['authorization'];
-    if (!authHeader) {
-        throw new AuthorizationError("Authorization Header is empty.");
+    try {
+        const authHeader = req.headers['authorization'];
+        if (!authHeader) {
+            throw new AuthorizationError("Authorization Header is empty.");
+        }
+        const authEmail = await getEmailFromAuth(authHeader);
+        const adminStatus = await isAdmin(authEmail);
+        return res.status(200).json({ isAdmin: adminStatus });
+    } catch (err) {
+        switch (err.name) {
+            case 'AuthorizationError':
+                console.error('AuthorizationError:', err);
+                return res.status(401).json({ message: err.message });
+            default:
+                console.error('Internal Server Error:', err);
+                return res.status(500).json({ message: 'Internal Server Error' });
```

Comment:
Same comment as before, but for the sake of time and getting this ticket completed, please just open a ticket to refactor the individual try catches to a generic error handling middleware